mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -0,0 +1,479 @@
+    else:
+        # skip the test for non-supported layouts.
+        # MIOpenDriver currently only support NCHW.
+        nanoSeconds = 0
+    output(config.generateCSVContent(nanoSeconds), outputFile)
+
+def printPerformance(mlirFileName, miopenFileName):
+    mlirOutput = open(mlirFileName, 'r')
+    mlirResults = [line[:-1] for line in mlirOutput.readlines()]
+ 
+    miopenOutput = open(miopenFileName, 'r')
+    miopenFlops = [','+line.rstrip().split(',')[-1] for line in miopenOutput.readlines()]
+
+    mlirOutput.close()
+    miopenOutput.close()
+
+    results = [ i+j for i, j in zip(mlirResults, miopenFlops)]
+    header = ConvConfiguration.generateCSVHeader()+',MIOpenTFlops'
+    
+    table = PrettyTable()
+    table.field_names = header.split(',')
+    for res in results:
+        table.add_row(res.split(','))
+    print(table)
+
+    htmlOutput = open("MLIR_vs_MIOpen.html", 'w')
+    htmlOutput.write(table.get_html_string(format=True, 
+                                           attributes={'border': 1, 
+                                                       'style': 'border-width:1px; border-collapse: collapse; background-color: #dddddd;'}))
+    htmlOutput.close()
+ 
+def generatePerformanceResults(configs, xdlops):
+    mlirFileName = date.today().strftime("mlir.%m%d%y")
+    mlirOutput = open(mlirFileName, 'w')
+    for testVector in configs:
+        benchmarkMLIR(testVector.split(sep=' '), mlirOutput, xdlops)
+
+    miopenFileName = date.today().strftime("miopen.%m%d%y")
+    miopenOutput = open(miopenFileName, 'w')
+    for testVector in configs:
+        benchmarkMIOpen(testVector.split(sep=' '), miopenOutput, xdlops)
+
+    mlirOutput.close()
+    miopenOutput.close()
+    printPerformance(mlirFileName, miopenFileName)     
+
+# Main function.
+if __name__ == '__main__':
+    """ 
+usage examples: 
+  python3 MIOpenDriver.py
+  python3 MIOpenDriver.py -o mlir.perf -b
+  python3 MIOpenDriver.py -o miopen.perf -bmiopen
+  python3 MIOpenDriver.py -o mlir.perf conv -F 1 -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
+  python3 MIOpenDriver.py -o miopen.perf -miopen conv -F 1 -n 256 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -t 1
+    """    
+
+    xdlops = False
+    configs = getConfigurations(configurationFileName);
+    r = subprocess.run(f'/opt/rocm/bin/rocm_agent_enumerator -t GPU|grep gfx908', shell=True)
+    if r.returncode == 0:
+        xdlops = True
+
+    if len(sys.argv) == 1:
+        # batch benchmark with MLIR and MIOpen.
+        generatePerformanceResults(configs, xdlops)
+    else:
+        outputFile = None
+        if sys.argv[1] == '-o':
+            outputFile = open(sys.argv[2], 'w')
+            fileName = sys.argv[2]
+            sys.argv.pop(1)
+            sys.argv.pop(1)
+
+        output(ConvConfiguration.generateCSVHeader(), outputFile)
+        if sys.argv[1] == '-b':
```

Comment:
This doesn't allow passing multiple arguments - is that intentional?

Reply:
Not sure what the question is. The comment block lists the valid arguments.

Reply:
(The question is expanded on in another comment)